wdl-lint/src/rules/missing_runtime.rs

```diff
@@ -75,9 +76,13 @@ impl Visitor for MissingRuntimeRule {
             return;
         }
 
-        if task.runtimes().next().is_none() {
-            let name = task.name();
-            state.add(missing_runtime_section(name.as_str(), name.span()));
+        // This rule should only be present for WDL v1.1 or earlier, as the
+        // `requirements` section replaces it in WDL v1.2.
+        if let SupportedVersion::V1(minor_version) = self.0.expect("version should exist here") {
+            if minor_version <= V1::One && task.runtimes().next().is_none() {
+                let name = task.name();
+                state.add(missing_runtime_section(name.as_str(), name.span()));
+            }
         }
```

Comment:
We need additional checks. One to warn that `runtime` is deprecated in 1.2+ documents. We also need a lint rule to warn on missing `requirements` sections in 1.2+ documents. Another check if `runtime` and `requirements` are specified in 1.2+ documents, though that probably ought to be a validation error if it isn't currently.

Some of that could be implemented here or in new rules, but we at least need to record that these are needed.

Reply:
I've added a new `MissingRequirements` rule to address this. This rule also flags `runtime` sections used when `requirements` should be. The validation takes care of both of them not being present today.